ChatSecure/Classes/Model/Yap Storage/OTRXMPPRoomOccupant.swift

```diff
@@ -93,10 +93,12 @@ open class OTRXMPPRoomOccupant: OTRYapDatabaseObject, YapDatabaseRelationshipNod
     
     @objc open static let roomEdgeName = "OTRRoomOccupantEdgeName"
     
-    @objc open var available = false
+    @objc open var available:Bool {
+        return (jids?.count ?? 0) > 0
+    }
     
-    /** This is the JID of the participant as it's known in the room i.e. baseball_chat@conference.dukgo.com/user123 */
-    @objc open var jid:String?
+    /** This is all JIDs of the participant as it's known in the room i.e. baseball_chat@conference.dukgo.com/user123 */
+    @objc open var jids:[String]?
```

Comment:
I think it would be better to only expose a computed `Set<XMPPJID>` instead of an array of strings, but keep this variable as private internal storage. For example:

```swift
    @objc private var _jids: [String]?
    
    @objc open var jids: Set<XMPPJID> {
        get {
            let validJids = _jids?.flatMap({ (jidStr) -> XMPPJID? in
                XMPPJID(string: jidStr)
            })
            return Set(validJids ?? [])
        }
        set {
            _jids = newValue.map { (jid) -> String in
                jid.full
            }
        }
    }
```